ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoicePayloadExecutorTest.java

```diff
@@ -102,6 +102,22 @@ void optimisticallyExecute_shouldValidateMergeBlockWhenThisIsTheMergeBlock() {
     assertThat(result).isTrue();
   }
 
+  @Test
+  void optimisticallyExecute_shouldReturnFailedExecutionOnMergeBlockWhenELOffline() {
+    when(executionEngine.getPowBlock(payload.getParentHash()))
+        .thenReturn(SafeFuture.failedFuture(new Error()));
+    final ForkChoicePayloadExecutor payloadExecutor = createPayloadExecutor();
+    final boolean execution = payloadExecutor.optimisticallyExecute(defaultPayloadHeader, payload);
+
+    // Should defer execution until it has checked the terminal difficulty so we expect getPoWBlock
+    verify(executionEngine).getPowBlock(payload.getParentHash());
+    verify(executionEngine, never()).executePayload(payload);
+    assertThat(execution).isTrue();
+    assertThat(payloadExecutor.getExecutionResult()).isCompleted();
+    final ExecutePayloadResult result = payloadExecutor.getExecutionResult().getImmediately();
+    assertThat(result.hasFailedExecution()).isTrue();
```

Comment:
this can be written using `isCompletedWithValueMatching`

Reply:
Nice point, have not discovered it